src/test/java/com/artipie/http/TrafficMetricSliceTest.java

```diff
@@ -0,0 +1,108 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.http;
+
+import com.artipie.http.rq.RequestLine;
+import com.artipie.http.rq.RqMethod;
+import com.artipie.http.rs.RsWithBody;
+import com.artipie.metrics.Metrics;
+import com.artipie.metrics.memory.InMemoryMetrics;
+import com.artipie.metrics.publish.MetricsOutput;
+import hu.akarnokd.rxjava2.interop.CompletableInterop;
+import io.reactivex.Flowable;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test case for {@link TrafficMetricSlice}.
+ *
+ * @since 0.10
+ */
+final class TrafficMetricSliceTest {
+
+    /**
+     * Request metrics key.
+     */
+    private static final String KEY_REQ = "request.body.size";
+
+    /**
+     * Response metrics key.
+     */
+    private static final String KEY_RSP = "response.body.size";
+
+    @Test
+    void updatesMetrics() throws Exception {
+        final Metrics metrics = new InMemoryMetrics();
+        final int size = 100;
+        new TrafficMetricSlice((line, head, body) -> new RsWithBody(body), metrics)
+            .response(
+                new RequestLine(RqMethod.GET, "/foo").toString(),
+                Collections.emptyList(),
+                Flowable.just(ByteBuffer.wrap(new byte[size]))
+            )
+            .send(
+                (status, headers, body) ->
+                    Flowable.fromPublisher(body).ignoreElements().to(CompletableInterop.await())
+            ).toCompletableFuture().get();
+        final AtomicLong request = new AtomicLong();
+        final AtomicLong response = new AtomicLong();
+        metrics.publish(
+            // @checkstyle AnonInnerLengthCheck (22 lines)
+            new MetricsOutput() {
+                @Override
+                public void counters(final Map<String, Long> data) {
+                    if (data.containsKey(TrafficMetricSliceTest.KEY_REQ)) {
+                        request.addAndGet(data.get(TrafficMetricSliceTest.KEY_REQ));
+                    }
+                    if (data.containsKey(TrafficMetricSliceTest.KEY_RSP)) {
+                        response.addAndGet(data.get(TrafficMetricSliceTest.KEY_RSP));
+                    }
+                }
+
+                @Override
+                public void gauges(final Map<String, Long> data) {
+                    if (data.containsKey(TrafficMetricSliceTest.KEY_REQ)) {
+                        request.set(data.get(TrafficMetricSliceTest.KEY_REQ));
+                    }
+                    if (data.containsKey(TrafficMetricSliceTest.KEY_RSP)) {
+                        response.set(data.get(TrafficMetricSliceTest.KEY_RSP));
+                    }
```

Comment:
@g4s8 it seems that this code block is not needed and so may be removed. It would be even better to throw an exception here, because we do not expect the slice to update any gauges.